framework/auth/views.py

```diff
@@ -35,6 +36,21 @@
 from osf.utils.requests import check_select_for_update
 from osf import features
 
+
+def throttle_password_reset(user):
+    if not throttle_period_expired(user.submit_reset_token_last_attempt, settings.TIME_RESET_SUBMIT_PASSWORD_TOKEN):
+        last_attempt = user.submit_reset_token_last_attempt.replace(tzinfo=pytz.utc)
+        time_since_throttle = (timezone.now() - last_attempt).total_seconds()
+        wait_time = int(settings.TIME_RESET_SUBMIT_PASSWORD_TOKEN - time_since_throttle)
+        error_data = {
+            'message_short': 'Too Many Requests.',
+            'message_long': 'The you have entered an incorrect token, please wait {} seconds to retry.'.format(wait_time)
+        }
+        raise HTTPError(429, data=error_data)
+    else:
+        user.submit_reset_token_last_attempt = timezone.now()
+        user.save()
```

Comment:
The user object is already being saved further down in `reset_password_post` when `verification_key_v2` is set, wonder if there's a way to potentially avoid a double user save in this method.

Reply:
I tried putting this in an atomic transaction and a few other things and this is the best option I can see.